enigma-principal/app/src/main.rs

```diff
@@ -43,27 +46,16 @@ mod esgx;
 
 fn main() {
     let opt: Opt = Opt::from_args();
-    println!("CLI params: {:?}", opt);
+    debug!("CLI params: {:?}", opt);
 
     let datadir = dirs::home_dir().unwrap().join(".enigma");
     let loggers = logging::get_logger(opt.debug_stdout, datadir.clone(), opt.verbose).expect("Failed Creating the loggers");
     CombinedLogger::init(loggers).expect("Failed initializing the logger");
 
-    // init enclave
-    let enclave = match esgx::general::init_enclave_wrapper() {
-        Ok(r) => {
-            println!("[+] Init Enclave Successful {}!", r.geteid());
-            r
-        }
-        Err(x) => {
-            println!("[-] Init Enclave Failed {}!", x.as_str());
-            return;
-        }
-    };
-
-    // run THE app
+    let enclave = esgx::general::init_enclave_wrapper().expect("[-] Init Enclave Failed");
```

Comment:
This `main` is very similar to [that](https://github.com/enigmampc/enigma-core/blob/develop/enigma-core/app/src/main.rs#L19). This is not the only place where the code is copied for KM node. I think it worth time to go over the code and extract all the duplicate code to the common services **ASAP**. Otherwise, we will get all the problems of the duplicate code (http://www.informit.com/articles/article.aspx?p=457502&seqNum=5)

Reply:
The principal node was initially all copied code. Of course I'm aware that code duplication is bad, but I don't feel comfortable modifying common components in this particular project. Fortunately, Elichai does and he did refactor most of the duplicate code into `enigma-tools-m`. The main still file has duplicated instructions but it's largely limited to initializing shared services. I would love to remove those duplicate instructions but I suggest that we coordinate first discuss verbally and design this together.